src/planner/binder.cpp

```diff
@@ -41,55 +42,76 @@ Binder::Binder(bool, ClientContext &context, shared_ptr<Binder> parent_p, bool i
-	case StatementType::EXECUTE_STATEMENT:
-		return Bind((ExecuteStatement &)statement);
-	case StatementType::LOGICAL_PLAN_STATEMENT:
-		return Bind((LogicalPlanStatement &)statement);
-	default: // LCOV_EXCL_START
-		throw NotImplementedException("Unimplemented statement type \"%s\" for Bind",
-		                              StatementTypeToString(statement.type));
-	} // LCOV_EXCL_STOP
+	try {
+		switch (statement.type) {
+		case StatementType::SELECT_STATEMENT:
+			return Bind((SelectStatement &)statement);
+		case StatementType::INSERT_STATEMENT:
+			return Bind((InsertStatement &)statement);
+		case StatementType::COPY_STATEMENT:
+			return Bind((CopyStatement &)statement);
+		case StatementType::DELETE_STATEMENT:
+			return Bind((DeleteStatement &)statement);
+		case StatementType::UPDATE_STATEMENT:
+			return Bind((UpdateStatement &)statement);
+		case StatementType::RELATION_STATEMENT:
+			return Bind((RelationStatement &)statement);
+		case StatementType::CREATE_STATEMENT:
+			return Bind((CreateStatement &)statement);
+		case StatementType::DROP_STATEMENT:
+			return Bind((DropStatement &)statement);
+		case StatementType::ALTER_STATEMENT:
+			return Bind((AlterStatement &)statement);
+		case StatementType::TRANSACTION_STATEMENT:
+			return Bind((TransactionStatement &)statement);
+		case StatementType::PRAGMA_STATEMENT:
+			return Bind((PragmaStatement &)statement);
+		case StatementType::EXPLAIN_STATEMENT:
+			return Bind((ExplainStatement &)statement);
+		case StatementType::VACUUM_STATEMENT:
+			return Bind((VacuumStatement &)statement);
+		case StatementType::SHOW_STATEMENT:
+			return Bind((ShowStatement &)statement);
+		case StatementType::CALL_STATEMENT:
+			return Bind((CallStatement &)statement);
+		case StatementType::EXPORT_STATEMENT:
+			return Bind((ExportStatement &)statement);
+		case StatementType::SET_STATEMENT:
+			return Bind((SetStatement &)statement);
+		case StatementType::LOAD_STATEMENT:
+			return Bind((LoadStatement &)statement);
+		case StatementType::EXTENSION_STATEMENT:
+			return Bind((ExtensionStatement &)statement);
+		case StatementType::PREPARE_STATEMENT:
+			return Bind((PrepareStatement &)statement);
+		case StatementType::EXECUTE_STATEMENT:
+			return Bind((ExecuteStatement &)statement);
+		case StatementType::LOGICAL_PLAN_STATEMENT:
+			return Bind((LogicalPlanStatement &)statement);
+		default: // LCOV_EXCL_START
+			auto &config = DBConfig::GetConfig(context);
+
+			for (auto &extension_op : config.operator_extensions) {
+				auto bound_statement = extension_op.Bind(context, *this, extension_op.operator_info.get(), statement);
+				if (bound_statement.plan != nullptr)
+					return bound_statement;
+			}
+
+			throw NotImplementedException("Unimplemented statement type \"%s\" for Bind",
+			                              StatementTypeToString(statement.type));
+		} // LCOV_EXCL_STOP
+	} catch (const StandardException &e) {
+		auto &config = DBConfig::GetConfig(context);
```

Comment:
We generally try to avoid `try/catch` everywhere outside of at the "root" of query execution. Can this be removed? The extension should only throw an exception if there is an unrecoverable error (i.e. it failed to bind its own custom node) - in which case we want to propagate the exception upwards anyway.

Reply:
The primary reason for adding the try-catch block was to call the extension's bind functions when duckdb Binding fails. Realize on second glance that using the existing exception block in https://github.com/duckdb/duckdb/blob/c0a4ab96c626426961c207f49c19aa81448e91da/src/planner/planner.cpp#L49 is probably a better location.

Reply:
Perhaps I'm misunderstanding but what is the use case for wanting to call the extensions' bind function when DuckDB's  bind function fails? Isn't that what the call to the extensions' bind in the switch statement is for?

Reply:
I will remove the call insight the switch as it is confusing and I don't have a good case for it. The call in the switch is only triggered by unhandled statement types. I can see this being useful if the flow of the parser extension changes, so that it returns a SQLStatement that is parsed by DuckDB. However, as it stands there should not be unhandled statement types.

With the try/catch I intend to capture catalog misses & other bind exceptions (StandardException - that fail a query, but don't kill a transaction). So 

   `insert into remote_foo values(1,2)` will trigger the call inside the catch if remote_foo is not a member of duckdb's catalog.

Reply:
Aha, that makes sense. I'm fine with that at least as a temporary solution - but I think a cleaner solution would be implementing the extensible catalog & pluggable storage systems (although clearly a lot more work). Are there other use cases you have for this - or is it limited to only emulating a pluggable catalog for now?